prql-compiler/src/sql/anchor.rs

```diff
@@ -66,10 +66,17 @@ impl AnchorContext {
         let def = self.columns_defs.get_mut(cid).unwrap();
 
         if def.name.is_none() {
-            let id = self.next_col_name_id;
-            self.next_col_name_id += 1;
+            match &def.expr.kind {
+                ExprKind::ExternRef { variable, .. } if variable == "*" => {
+                    def.name = Some("*".into());
```

Comment:
This is clever but a bit hacky. Passing a `"*"` does indeed prevent alias to be emitted, because it uses rule that SQL assumes alias from ident names only, so we omit explicit alias.

If we don't want to rely on this rule, I have to do a little refactor.